src/lib.rs

```diff
@@ -367,20 +367,13 @@ compile_error!(
      `async-std-runtime` or set `default-features = false` in your Cargo.toml"
 );
 
-#[cfg(all(feature = "tokio-runtime", feature = "sync"))]
+#[cfg(all(feature = "async-std-runtime", feature = "sync"))]
 compile_error!(
-    "`tokio-runtime` and `sync` can't both be enabled; either disable `sync` or set \
-     `default-features = false` in your Cargo.toml"
+    "`async-std-runtime` and `sync` can't both be enabled"
```

Comment:
I was making the code changes for this but think doing this will cause breaking changes as well... currently, the `sync` feature in our `Cargo.toml` enables `async-std-runtime`, and I swapped that out for `tokio-runtime` for this implementation. But if we want to allow for either, I think we'd want something like
```
[features]
...
sync = []
```
in `Cargo.toml` so as not to enable one runtime feature when the other is intended for use. However, this will break compilation for users who currently have
```
mongodb = { version = "2.1.0", default-features = false, features = ["sync"] }
```
in their `Cargo.toml` as they'll no longer have a runtime specified.

Is there a clean way around this that I'm missing? If we leave `async-std-runtime` enabled for `sync` then users will have to pull in both runtimes when they want to use tokio which isn't great. Alternatively we could make a `tokio-sync` feature for using sync with tokio and clean this up if we ever make breaking changes down the line.

Reply:
Ah good point. Hm, I'm not sure there _is_ a good option here besides the `tokio-sync` feature you suggest, since if we have any runtime in the `sync = [...]` definition, then the other won't be able to be used. This would present issues for RUST-488, since we'd ideally want to allow users to use whichever runtime they want.

So yeah, I guess our choices are the following:
- have `sync = ["tokio-runtime"]` per this PR, but don't allow both `sync` and `async-std-runtime` when we do RUST-488
    - pros: no new feature flag, sync users get tokio under the hood automatically
    - cons: technically breaking change, no async-std-runtime + sync
- introduce `tokio-sync` = ["tokio-runtime"]
    - pros: can use sync API with either tokio or async-std once we do RUST-488, no breaking changes
    - cons: new feature flag to maintain, existing sync users don't get benefits

I think I personally lean towards the new flag to avoid any potential (albeit rare) breaking changes and also to avoid another mutual exclusion problem between feature flags. I guess we'd probably want `sync` and `tokio-sync` to be mutually exclusive, or we could just use tokio if they're both enabled.

WDYT? cc @abr-egn

Reply:
Agreed that the `tokio-sync` flag seems like the least bad option here; it seems particularly worth avoiding the "can't use async-std-runtime + sync" problem.

Reply:
updated to introduce a `tokio-sync` feature flag